src/geometry/PolySetBuilder.h

```diff
@@ -0,0 +1,25 @@
+#pragma once
+
+
+#include "Geometry.h"
+#include "linalg.h"
+#include "GeometryUtils.h"
+#include "Polygon2d.h"
+#include "boost-utils.h"
+
+#include <vector>
+#include <string>
+#include <unordered_map>
+#include <hash.h>
+class PolySet;
+
+class PolySetBuilder
+{
+  PolySet *ps;
+  std::unordered_map<Vector3d, int> pointMap;
```

Comment:
FYI: Reindexer already solves this for any type, including Vector3d. Might not be super important right now, but could be interesting if we were to add more vertex attributes in the future.